array/bool.go

```diff
@@ -6,3 +6,14 @@ type Boolean interface {
 	Value(i int) bool
 	BooleanSlice(start, stop int) Boolean
 }
+
+// BooleanBuilder represents an abstraction over building a bool array.
+type BooleanBuilder interface {
+	BaseBuilder
+	Append(v bool)
+	AppendNull()
```

Comment:
nit: you could move `AppendNull` to the `BaseBuilder` interface, as it is not specific